packages/uploadthing/src/sdk/index.ts

```diff
@@ -219,7 +219,19 @@ export class UTApi {
     const filesToUpload = await Promise.all(
       asArray(urls).map(async (_url) => {
         let url = isObject(_url) ? _url.url : _url;
-        if (typeof url === "string") url = new URL(url);
+
+        if (typeof url === "string") {
+          // since dataurls will result in name being too long, tell the user
+          // to use uploadFiles instead.
+          if (url.startsWith("data:")) {
+            throw new UploadThingError({
+              code: "BAD_REQUEST",
+              message:
+                "Please use uploadFiles() for data URLs. uploadFilesFromUrl() is intended for use with remote URLs only.",
+            });
```

Comment:
wait should we throw? the inner function returns either data or error and let's the other continue

Reply:
We throw below if ant of the files fail to download. Possibly we should not throw in either case then...

Reply:
right... we should probably just make sure to add an error for that place in the final result array?